sqlglot/parser.py

```diff
@@ -903,28 +888,52 @@ def _parse_column(self):
 
         return self._parse_brackets(this)
 
-    def _parse_function(self, this, schema=False):
-        if not this:
-            return this
-        if this.token_type == TokenType.CASE:
+    def _parse_function(self, schema=False):
+        if self._match(TokenType.CASE):
             return self._parse_case()
-        if not self._match(TokenType.L_PAREN):
-            return this
 
-        if this.token_type == TokenType.CAST:
+        if self._match(TokenType.CAST):
+            self._match(TokenType.L_PAREN)
             this = self._parse_cast()
-        elif this.token_type == TokenType.COUNT:
+            self._match(TokenType.R_PAREN)
```

Comment:
Here and in branches below: should we actually check whether the match was successful?

Reply:
Wouldn't the lack of right parenthesis be considered a syntax error?